opfunu/utils/operator.py

```diff
@@ -232,30 +257,51 @@ def lunacek_bi_rastrigin_func(x, z, miu0=2.5, d=1.):
     return result1 + 10 * (ndim - np.sum(np.cos(2 * np.pi * z)))
 
 
-def calculate_weight(x, xichma=1.):
+def calculate_weight(x, delta=1.):
     ndim = len(x)
-    weight = 1
     temp = np.sum(x ** 2)
     if temp != 0:
-        weight = (1.0 / np.sqrt(temp)) * np.exp(-temp / (2 * ndim * xichma ** 2))
+        weight = np.sqrt(1.0 / temp) * np.exp(-temp / (2 * ndim * delta ** 2))
+    else:
+        weight = 1e99  # this is the INF definition in original CEC Calculate logic
+
     return weight
 
 
 def modified_schwefel_func(x):
-    x = np.array(x).ravel()
-    ndim = len(x)
-    z = x + 4.209687462275036e+002
-    return 418.9829 * ndim - np.sum(gz_func(z))
+    """
+        This is a direct conversion of the CEC2021 C-Code for the Modified Schwefel F11 Function
+    """
+    z = np.array(x).ravel() + 4.209687462275036e+002
+    nx = len(z)
+
+    mask1 = z > 500
+    mask2 = z < -500
+    mask3 = ~mask1 & ~mask2
+    fx = np.zeros(nx)
+    fx[mask1] -= (500.0 + np.fmod(np.abs(z[mask1]), 500)) * np.sin(np.sqrt(500.0 - np.fmod(np.abs(z[mask1]), 500))) - (
+                (z[mask1] - 500.0) / 100.) ** 2 / nx
+    fx[mask2] -= (-500.0 + np.fmod(np.abs(z[mask2]), 500)) * np.sin(np.sqrt(500.0 - np.fmod(np.abs(z[mask2]), 500))) - (
+                (z[mask2] + 500.0) / 100.) ** 2 / nx
+    fx[mask3] -= z[mask3] * np.sin(np.sqrt(np.abs(z[mask3])))
+
+    return np.sum(fx) + 4.189828872724338e+002 * nx
+
 
 
 def happy_cat_func(x):
-    x = np.array(x).ravel()
-    ndim = len(x)
-    t1 = np.sum(x)
-    t2 = np.sum(x ** 2)
+    z = np.array(x).ravel()
+    ndim = len(z)
+    t1 = np.sum(z)
+    t2 = np.sum(z ** 2)
     return np.abs(t2 - ndim) ** 0.25 + (0.5 * t2 + t1) / ndim + 0.5
 
 
+def happy_cat_shifted_func(x):
```

Comment:
@firestrand,

Could you please add a 'shifted' parameter to this type of function? Kindly avoid fixed the shifted value within the function. This will make the function more reusable.